pkg/proxy/roundrobin.go

```diff
@@ -210,14 +236,14 @@ func (lb *LoadBalancerRR) OnUpdate(endpoints []api.Endpoints) {
 	for _, endpoint := range endpoints {
 		existingEndpoints, exists := lb.endpointsMap[endpoint.Name]
 		validEndpoints := filterValidEndpoints(endpoint.Endpoints)
-		if !exists || !reflect.DeepEqual(existingEndpoints, validEndpoints) {
+		if !exists || !endpointsAreEqual(existingEndpoints, validEndpoints) {
```

Comment:
Now you have pessimized - validEndpoints is wholly owned at this point.  Can we do something like:
```
// We own validEndpoints but existingEndpoints is a reference to the stored data.
if !exists || !reflect.DeepEqual(util.SortSlice(util.CopySlice(existingEndpoints)), util.SortSlice(validEndpoints)) {
```

I am dismayed by Go's sharp edges and lack of helpful functions here.